repos/public-user/src/Module/AclRules.php

```diff
@@ -0,0 +1,181 @@
+use Omeka\Permissions\Assertion\OwnsEntityAssertion;
+use Omeka\Permissions\Assertion\SiteIsPublicAssertion;
+use Psr\Container\ContainerInterface;
+use PublicUser\Acl\IsOnRouteAssertion;
+use PublicUser\Acl\IsRegistrationPermittedAssertion;
+use PublicUser\Acl\UserHasSitePermission;
+use PublicUser\Controller\AccountController;
+use PublicUser\Controller\AuthController;
+use PublicUser\Controller\LoginController;
+use PublicUser\Controller\PublicProfileController;
+use PublicUser\Controller\SiteLoginRedirectController;
+use PublicUser\Settings\PublicUserSettings;
+use PublicUser\Site\SiteProvider;
+use Zend\Permissions\Acl\Acl;
+use Zend\Permissions\Acl\Assertion\AssertionAggregate;
+use Zend\Router\RouteInterface;
+
+trait AclRules
+{
+    /**
+     * @return ContainerInterface
+     */
+    abstract public function getServiceLocator();
+
+    private function addAclRules(RouteInterface $router, $request)
+    {
+        /** @var Acl $acl */
+        $acl = $this->getServiceLocator()->get('Omeka\Acl');
+        /** @var PublicUserSettings $settings */
+        $settings = $this->getServiceLocator()->get(PublicUserSettings::class);
+        /** @var SiteProvider $siteProvider */
+        $siteProvider = $this->getServiceLocator()->get(SiteProvider::class);
+
+        // Site permissions
+        // The configuration for the extended site permissions (what the new roles can do) is added by the PublicUser
+        // and so the definitions for the site permissions are too. This is crossing over the modules a little bit,
+        // and they should live in the IIIF-Storage module.
+        $acl->addResource('iiif-collection');
+        $acl->addResource('iiif-manifest');
+        $acl->addResource('iiif-canvas');
+
+        $acl->allow(
+            null,
+            'iiif-collection',
+            'view-all',
+            new UserHasSitePermission(
+                $settings,
+                'view-all-collections'
+            )
+        );
+        $acl->allow(
+            null,
+            'iiif-collection',
+            'view',
+            new UserHasSitePermission(
+                $settings,
+                'view-collection'
+            )
+        );
+        $acl->allow(
+            null,
+            'iiif-manifest',
+            'view',
+            new UserHasSitePermission(
+                $settings,
+                'view-manifest'
+            )
+        );
+        $acl->allow(
+            null,
+            'iiif-canvas',
+            'view',
+            new UserHasSitePermission(
+                $settings,
+                'view-canvas'
+            )
+        );
+
+        // The default assertion for viewing resources.
+        $viewerAssertion = (new AssertionAggregate())
+            ->setMode(AssertionAggregate::MODE_AT_LEAST_ONE)
+            ->addAssertions([
+                new SiteIsPublicAssertion,
+                new HasSitePermissionAssertion('viewer'),
+            ]);
```

Comment:
This taken from Omeka's own ACL and is the default assertion for wether a particular user can see a site. This was needed to support non-public sites.